src/app/Rules/Rules.tsx

```diff
@@ -58,6 +60,27 @@ export interface Rule {
   maxSizeBytes: number;
 }
 
+export const parseRule = (file: File): Observable<Rule> => {
+  return from(
+    file.text()
+    .then(content => {
+      const ruleConfig = JSON.parse(content);
+      const rule: Rule = {
+        name: ruleConfig.name,
+        description: ruleConfig.description,
+        matchExpression: ruleConfig.matchExpression,
+        eventSpecifier: ruleConfig.eventSpecifier,
+        archivalPeriodSeconds: ruleConfig.archivalPeriodSeconds,
+        preservedArchives: ruleConfig.preservedArchives,
+        maxAgeSeconds: ruleConfig.maxAgeSeconds,
+        maxSizeBytes: ruleConfig.maxSizeBytes
+      };
```

Comment:
Is this explicit structuring of the `Rule` needed? It seems like this can probably be simplified by simply:

```typescript
export const parseRule = (file: File): Observable<Rule> => {
  return from(
    file.text().then(JSON.parse)
  );
};
```

Also, is this exported function only used in the new `Modal` component? If so, why not define it there?

Reply:
> Also, is this exported function only used in the new Modal component? If so, why not define it there?

Oh yes I was grouping them near Rule interface but I think it makes more sense.

> Is this explicit structuring of the Rule needed? It seems like this can probably be simplified by simply:

@andrewazores I wonder if the structure of the rule file needs to follow the exact structure of `Rule` or is there a case where a user want to define another field like units similar to UI?